### PR TITLE
Cleanup animation examples

### PR DESCRIPTION
**Pull request by anntzer**
I thought that the animation examples section was a bit of an unstructured mess so I tried to reorganize it somewhat.  The suggested new list of examples (open to discussion, of course) is

simple examples
===============
animation_demo: plt.pause example
simple_anim: FuncAnimation example
dynamic_image: FuncAnimation example
dynamic_image2: ArtistAnimation example

intermediate examples
=====================
strip_chart: FuncAnimation example

advanced examples
=================
histogram: animating a composite artist (namely, a histogram)

nice examples
=============
bayes: FuncAnimation example
double_pendulum: FuncAnimation example, skipped as needing scipy
random_walk (renamed from simple_3danim): FuncAnimation example
rain: FuncAnimation example
unchained: FuncAnimation example

movie writing
=============
frame_grabbing_sgskip (renamed from moviewriter_sgskip.py)

removed
=======
animate_decay: overlaps with simple_anim
basic_example: equivalent to {simple_anim + dynamic_image2}
basic_example_writer_sgskip: split as comments to
  simple_anim/dynamic_image/dynamic_image2
random_data: overlaps with simple_anim, though using a generator
subplots: relies on subclassing private attributes of TimedAnimation, would
  better be replaced by a FuncAnimation example, overly complex

-----

It would be nice if s-g allowed for an easier way to set the example order, but that's for another time...

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by tacaswell** [comment]
It is probably worth preserving an example of `FuncAnimation` that uses a generator.

**Comment by tacaswell** [comment]
I think it is fair to require scipy to build the docs (but not in the library).

File: examples/animation/simple_anim.py [hunk 1]
```diff
@@ -15,16 +15,27 @@
 line, = ax.plot(x, np.sin(x))
 
 
-def animate(i):
-    line.set_ydata(np.sin(x + i/10.0))  # update the data
+def init():  # only required for blitting to give a clean slate.
+    line.set_ydata(x)
```
Comment (tacaswell) [inline examples/animation/simple_anim.py]: This sets the initial data to be a line?

File: examples/animation/dynamic_image2.py [hunk 2]
```diff
@@ -30,6 +30,14 @@ def f(x, y):
...
 
-# ani.save('dynamic_images.mp4')
+# To save the animation, use e.g.
+#
+# ani.save("movie.mp4")
+#
+# or
+#
+# from matplotlib.animation import FFMpegWriter
+# writer = Writer(fps=15, metadata=dict(artist='Me'), bitrate=1800)
```
Comment (tacaswell) [inline examples/animation/dynamic_image2.py]: This should be `FFMpegWriter` or the above line should be `... as Writer`.  Same for other files.

File: examples/animation/animate_decay.py [hunk 3]
```diff
@@ -1,53 +0,0 @@
...
-def run(data):
-    # update the data
-    t, y = data
-    xdata.append(t)
-    ydata.append(y)
-    xmin, xmax = ax.get_xlim()
-
-    if t >= xmax:
-        ax.set_xlim(xmin, 2*xmax)
-        ax.figure.canvas.draw()
```
Comment (tacaswell) [inline examples/animation/animate_decay.py]: This one shows how to force a change to the limits in the update function, is that shown else where still?

**Comment by anntzer** [comment]
Re: scipy: I rewrote bayes to not use scipy (it's not the end of the world to write the beta pdf).  double pendulum needs an ode solver, but it's not as if s-g can show animations anyways so I think it's fine to leave it as sgskipped.

File: examples/animation/simple_anim.py (hunk 1 above)
Comment (anntzer) [reply examples/animation/simple_anim.py]: good catch, fixed

File: examples/animation/dynamic_image2.py (hunk 2 above)
Comment (anntzer) [reply examples/animation/dynamic_image2.py]: fixed

File: examples/animation/animate_decay.py (hunk 3 above)
Comment (anntzer) [reply examples/animation/animate_decay.py]: I decided to retsore the example (also showcasing the use of a generator as you mentioned above) and remove dynamic_image instead (renaming dynamic_image2 (ArtistAnimation) to it).

**Comment by dstansby** [comment]
(going to let someone else review and merge since it involves removing stuff from the docs, but I'm 👍 )

**Comment by dopplershift** [comment]
Of course, now there's a conflict.

**Comment by anntzer** [comment]
rebased

**Comment by dstansby** [comment]
Since life's too short to wait for a travis OSX build to complete on a doc PR, I'm going to merge this.

**Pull request merged**